### PR TITLE
Add Kubernetes Job support to Weave Scope

### PR DESCRIPTION
**Pull request by satyamz**
## This covers:
- Implementation of the Job kubernetes resource in probe
- Reporter changes for the Job
- Add Describe Control to the Job
- Pass Job object to the kube controller renderer

##### 1. Jobs in the controllers topology (In the square shape)
 
Screenshot from 2019-05-19 02-36-29

##### 2. Job weave scope native description
Screenshot from 2019-05-19 02-44-47

##### 3. Job control option, kubectl describe

Screenshot from 2019-05-19 02-45-20

### TODO:
- ~Need to decide the shape for the Job.~

fixes: #3362 
Signed-off-by: Satyam Zode <satyamzode@gmail.com>

### CODE REVIEW

**Comment by satyamz** [comment]
Hi @fbarl, @bia I could use your help on deciding shape for the `Job`. Any ideas? :smile:

**Comment by qiell** [comment]
Just an idea, we can have the `job` shape similar to `cronjob` but with dotted lines.

**Comment by bia** [comment]
That's a good idea, @qiell. 😄 
@fbarl and I talked through the options and we think that we should do exactly what you said : dotted triangles for `job`s.

**Comment by satyamz** [comment]
Thanks @qiell,  @bia, @fbarl !! 

@fbarl We are raising PR in `ui-components` for the above change. After that, can you please bump up the version here?

**Comment by fbarl** [comment]
> We are raising PR in `ui-components` for the above change.

That's awesome, thanks!

> After that, can you please bump up the version here?

Sure, you can also add me as a reviewer for your PR in `ui-components`.

File: report/report.go [hunk 1]
```diff
@@ -297,6 +302,10 @@ func MakeReport() Report {
 			WithTag(Camera).
 			WithLabel("volume snapshot data", "volume snapshot data"),
 
+		Job: MakeTopology().
+			WithShape(StorageSheet).
```
Comment (fbarl) [inline report/report.go]: Just a reminder to replace this with `DottedTriangle` once we get the new shape from `ui-components`.

Comment (satyamz) [reply report/report.go]: Yes!

**Comment by fbarl** [comment]
@qiell The new `DottedTriangle` shape should now be available if you bump `ui-components` to `v0.20.0` (https://github.com/weaveworks/ui-components/commits/master) - from what I can tell, none of the other changes in that repo will have an effect on Scope so it's safe to do it directly in this PR :+1:

**Comment by satyamz** [comment]
Hi @fbarl , @bia 

This is how it looks now. Thanks!!

image

**Comment by satyamz** [comment]
> Just one small thing before merging - could you please add the get verb to the batch API group for jobs in examples/k8s/cluster-role.yaml? Otherwise the describe action is forbidden when trying to run the examples.

Good catch @fbarl! I updated the examples.

**Pull request merged**